GVFS/GVFS.Common/Maintenance/GitMaintenanceStep.cs

```diff
@@ -143,6 +147,45 @@ protected EventMetadata CreateEventMetadata(Exception e = null)
             return metadata;
         }
 
+        protected bool EnoughTimeBetweenRuns()
+        {
+            if (!this.Context.FileSystem.FileExists(this.LastRunTimeFilePath))
+            {
+                return true;
+            }
+
+            string lastRunTime = this.Context.FileSystem.ReadAllText(this.LastRunTimeFilePath);
+            if (!long.TryParse(lastRunTime, out long result))
+            {
+                this.Context.Tracer.RelatedError("Failed to parse long: {0}", lastRunTime);

```

Comment:
Good question!

It will put an error in telemetry if the file exists and we can't read the timestamp.
Given this is a weird state I'd say skip the maintence, until we root cause (no worse than we have today).

@derrickstolee what are your thoughts on this?0

Reply:
Tracing the fail to parse and running anyway should be fine. It will replace the file, we check for the git processes to avoid it being too soon.

Reply:
Updated to return true